media_kit_video/linux/gl_render_thread.cc

```diff
@@ -0,0 +1,103 @@
+// This file is a part of media_kit
+// (https://github.com/media-kit/media-kit).
+//
+// Copyright © 2025 Predidit.
+// All rights reserved.
+// Use of this source code is governed by MIT license that can be found in the
+// LICENSE file.
+
+#include "include/media_kit_video/gl_render_thread.h"
```

Comment:
The implementation seems very identical to ThreadPool used on Windows.
Please check if it can be shared (common folder), if not, then make it a GTK/GLib API.